From 6b38675734f1ecbfb9936637cff142c3a37af38d Mon Sep 17 00:00:00 2001
From: "Sean P. Kelly" <seankell@amazon.com>
Date: Thu, 31 Oct 2024 20:32:09 +0000
Subject: [PATCH] sleep before starting plugins

---
 cmd/nvidia-device-plugin/main.go | 1 +
 1 file changed, 1 insertion(+)

diff --git a/cmd/nvidia-device-plugin/main.go b/cmd/nvidia-device-plugin/main.go
index 7424c7c..5f09121 100644
--- a/cmd/nvidia-device-plugin/main.go
+++ b/cmd/nvidia-device-plugin/main.go
@@ -205,6 +205,7 @@ func start(c *cli.Context, flags []cli.Flag) error {
 
 	klog.Info("Starting OS watcher.")
 	sigs := watch.Signals(syscall.SIGHUP, syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT)
+	time.Sleep(45 * time.Second)
 
 	var started bool
 	var restartTimeout <-chan time.Time
-- 
2.40.1

